magic.js

```diff
@@ -168,15 +157,18 @@ function rsaverify(digest, padding) {
    */
   function keying(key) {
     return new Promise((resolve, reject) => {
-      if (key.startsWith('-----BEGIN PUBLIC KEY-----'))     { return resolve(key); }
       if (key.startsWith('-----BEGIN RSA PUBLIC KEY-----')) { return resolve(key); }
+      if (key.startsWith('-----BEGIN RSA PRIVATE KEY-----')) { return resolve(key); }
```

Comment:
Why would we resolve with the private key? The purpose of this function is to return a public key either because a public key was provided (line 160) or because we extracted it from the Private key (lines 163-171 below)

Reply:
Since node 12 private keys may be passed to verify / publicEncrypt operations and the public key will be extracted out of them automatically by the underlying implementation. I'm keeping lines 160-161 as optimisation - the PEM label forces these to be checked to be pkcs1 which can only be id-rsaEncryption OID keys. The optimization is that these are passed directly to openssl rather than being instantiated as key object and subsequently passed around between contexts.

Reply:
I didn't bother with the purpose of the function because it is not part of the public API, it is defined in and consumed by rsaVerify.

Reply:
> Since node 12 private keys may be passed to verify / publicEncrypt operations and the public key will be extracted out of them automatically by the underlying implementation. 

Didn't know that, thanks for explaining.